### PR TITLE
image: duplicate version caching bugfix

### LINKED ISSUE #5674: duplicate of an image can have version number of 0
**Describe the bug**
It is possible to duplicate an image such that the version of the duplicate is 0.

**To Reproduce**
1. Select a freshly imported image
2. Go to darkroom view
3. Expand "duplicate manager"
4. Click "create a duplicate with same history stack"
5. Expand "image information" -- verify that you are now editing a duplicate with version 1
6. Click "create a duplicate with same history stack"
6. You are now editing a third duplicate. In "image information" the version is 0.

**Expected behavior**
The version of the second duplicate should be 2.

**Screenshots**
image

**Platform:**
 - Darktable Version: master aef614f0e3caa8bfff91559b7a50df24e4e7357d
 - OS: Debian bullseye/sid, Gnome 3.36.3
 - OpenCL: yes
 - Nvidia GP104GLM [Quadro P4000 Mobile] (rev a1), OpenGL version string: 4.6.0 NVIDIA 440.82

**Additional context**
 - Have reproduced with a CR2 and a JPEG
 - Have reproduced with an empty/new config-dir
 - This bug has been present at least for the past few weeks

### PR DESCRIPTION
**Pull request by dtorop**
Set duplicate image version in the DB before the cache entry for the image is first allocated. Otherwise, image version in the cache will not reflect the version in the DB.

Also, the change timestamp and group ID for the original image were being written.  Instead, update these for the duplicate.

Fixes #5674.

### CODE REVIEW

File: src/common/image.c [hunk 1]
```diff
@@ -935,9 +928,18 @@ int32_t dt_image_duplicate_with_version(const int32_t imgid, const int32_t newve
 
     g_free(filename);
 
-    const dt_image_t *img = dt_image_cache_get(darktable.image_cache, imgid, 'r');
+    const dt_image_t *img = dt_image_cache_get(darktable.image_cache, newid, 'r');
```
Comment (TurboGit) [inline src/common/image.c:931]: This looks wrong to me. You are setting the group-id to be the one of the new image (the duplicate itself). The group-id of the artifact should be the group of the main image.

Comment (dtorop) [reply src/common/image.c:931]: That makes sense. I'll fix/re-test...

Comment (TurboGit) [reply src/common/image.c:931]: Note that the issue may already be fixed with my latest work on duplicate.

Comment (dtorop) [reply src/common/image.c:931]: I'm just looking at this. This PR fixes something different, that the calls to dt_tag_detach_by_string() and dt_image_cache_unset_change_timestamp() will call dt_image_cache_unset_change_timestamp() which load the version # in from the DB. So it's important to set the version # before it gets cached. (Or an alternative would be to make the version # writeable via the cache.)

Also dt_image_cache_unset_change_timestamp() should be working on the new image, even if I was wrong about the group ID.

**Comment by dtorop** [comment]
I fixed https://github.com/darktable-org/darktable/pull/5678#discussion_r448252127 and rebased onto 08c1dcd756079487df79b300893dda7963ebc1f6. This PR should still be necessary to fix #5674, which I can otherwise replicate in current git master.

**Comment by TurboGit** [comment]
Ok, works for me. Thanks.

**Pull request merged**